pkg/components/pluggable.go

```diff
@@ -50,3 +62,8 @@ type Pluggable struct {
 	// Version is the pluggable component version.
 	Version string
 }
+
+// Category returns the component category based on its pluggable type.
+func (p Pluggable) Category() Category {
```

Comment:
What is the difference between Type and Category? I saw it being used interchangeably in the code. We should probably pick one if they mean the same thing.

/cc @ItalyPaleAle  @yaron2

Reply:
Basically, it differs only for bindings.

the way we reference bindings components regardless of being input or output is bindings.$component

so the bindings registry is the only one that has a function to check if the component implements input or/and outputbinding

however they have different registries, so when implementing a pluggable component it should be clear wether the component is input or output,

category maps from the given component “interface” to that component category, note that the category concept already exists it was basically moved to components package